lib/spack/spack/cmd/style.py

```diff
@@ -216,7 +221,7 @@ def print_tool_header(tool):
 def run_flake8(file_list, args):
     returncode = 0
     print_tool_header("flake8")
-    flake8_cmd = which("flake8", required=True)
+    flake8_cmd = get_executable('flake8', spec='py-flake8', install=True)
```

Comment:
This doesn't work for me on a stock ubuntu if I don't install setuptools first on the system:
```console
==> Installing py-setuptools-scm-4.1.2-rd7ecfn5tjaxiwpkdnrxe2znlyleccyl
==> No binary for py-setuptools-scm-4.1.2-rd7ecfn5tjaxiwpkdnrxe2znlyleccyl found: installing from source
==> Using cached archive: /home/culpo/PycharmProjects/spack/var/spack/cache/_source-cache/archive/a8/a8994582e716ec690f33fec70cca0f85bd23ec974e3f783233e4879090a7faa8.tar.gz
==> py-setuptools-scm: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 1:
    '/usr/bin/python3.6' '-s' 'setup.py' '--no-user-cfg' 'build'
See build log for details:
  /tmp/culpo/spack-stage/spack-stage-py-setuptools-scm-4.1.2-rd7ecfn5tjaxiwpkdnrxe2znlyleccyl/spack-build-out.txt
```
This means installing `python3` and `python3-setuptools` with apt. If I install `python3-setuptools` I get a different error:
```
Traceback (most recent call last):
  File "/home/culpo/PycharmProjects/spack/opt/spack/linux-ubuntu18.04-broadwell/gcc-10.1.0/py-flake8-3.8.2-sprazagnhxqodvnn6mzfwtkheligd24v/bin/flake8", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3088, in <module>
    @_call_aside
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3072, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 3101, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 574, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 892, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 778, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'flake8==3.8.2' distribution was not found and is required by the application
==> Error: Flake8 style checks found errors
==> Error: style: mypy is not available in path, skipping
spack style found errors.
```